src/providers/Auth.js

```diff
@@ -0,0 +1,22 @@
+import React, { useState } from "react"
+
+const AuthContext = React.createContext()
+
+function AuthProvider(props) {
+  const [user, setUser] = useState({})
+
+  const login = () => {} // make a login request
+  const register = () => {} // register the user
+  const logout = () => {} // clear the token in localStorage and the user data
```

Comment:
You can now add some login in those functions eg inside login we can setUser({ ...mock a user here... }) 
and inside logout we can delete the user. eg setUser(undefined)

Reply:
Hello, good job. Are you building this from scratch or using the auth0 ?